src/core_codemods/semgrep/semgrep_nan_injection.py

```diff
@@ -0,0 +1,122 @@
+from textwrap import dedent
+
+import libcst as cst
+from libcst.codemod import ContextAwareVisitor
+
+from codemodder.codemods.base_codemod import (
+    Metadata,
+    ReviewGuidance,
+    ToolMetadata,
+    ToolRule,
+)
+from codemodder.codemods.base_visitor import UtilsMixin
+from codemodder.codemods.libcst_transformer import (
+    LibcstResultTransformer,
+    LibcstTransformerPipeline,
+)
+from codemodder.codemods.semgrep import SemgrepSarifFileDetector
+from core_codemods.semgrep.api import SemgrepCodemod, semgrep_url_from_id
+
+
+class NanInjectionTransformer(LibcstResultTransformer):
+    change_description = "Wrap typecast in if statement to check for `nan`"
```

Comment:
I'm not sure "wrap" is the right verb here since that has a more specific connotation (imo).
```suggestion
    change_description = "Add validation to untrusted numerical input to disallow `nan`"
```